File: day2_part2_2015.c
#include <stdio.h>

int main() {
    FILE *file = fopen("input.txt", "r");
    int l, w, h;
    int total_paper = 0;
    int total_ribbon = 0;

    while (fscanf(file, "%dx%dx%d", &l, &w, &h) != EOF) {
        int side1 = l * w;
        int side2 = w * h;
        int side3 = h * l;
        int smallest_side = side1 < side2 ? (side1 < side3 ? side1 : side3) : (side2 < side3 ? side2 : side3);

        total_paper += 2*l*w + 2*w*h + 2*h*l + smallest_side;
        total_ribbon += 2 * (l + w + h - (l > w ? (l > h ? l : h) : (w > h ? w : h))) + l * w * h;
    }

    printf("%d\n", total_ribbon);

    fclose(file);
    return 0;
}

File: objectivec/day2_part2_2015.m
#import <Foundation/Foundation.h>

int main(int argc, const char * argv[]) {
    @autoreleasepool {
        FILE *file = fopen("input.txt", "r");
        if (file == NULL) {
            printf("Error opening the file\n");
            return 1;
        }
        
        int l, w, h;
        int totalPaper = 0;
        int totalRibbon = 0;
        
        while (fscanf(file, "%dx%dx%d\n", &l, &w, &h) != EOF) {
            int side1 = l * w;
            int side2 = w * h;
            int side3 = h * l;
            int slack = MIN(MIN(side1, side2), side3);
            
            totalPaper += 2*l*w + 2*w*h + 2*h*l + slack;
            totalRibbon += 2*l + 2*w + 2*h - 2*MAX(MAX(l, w), h) + l*w*h;
        }
        
        printf("Total square feet of wrapping paper: %d\n", totalPaper);
        printf("Total feet of ribbon: %d\n", totalRibbon);
        
        fclose(file);
    }
    return 0;
}